daemon/containerd/service.go

```diff
@@ -127,21 +128,58 @@ func (i *ImageService) StorageDriver() string {
 	return i.snapshotter
 }
 
+// ImageDiskUsage returns the number of bytes used by content and layer stores
+// called from disk_usage.go
+func (i *ImageService) ImageDiskUsage(ctx context.Context) (int64, error) {
+	diskUsage, err := i.layerDiskUsage(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	// Include the size of content size from the images.
+	imgs, err := i.images.List(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	visitedImages := make(map[digest.Digest]struct{})
+	for _, img := range imgs {
+		if _, ok := visitedImages[img.Target.Digest]; ok {
+			continue
+		}
+		visitedImages[img.Target.Digest] = struct{}{}
+
+		if err = i.walkReachableImageManifests(ctx, img, func(imageManifest *ImageManifest) error {
+			size, err := imageManifest.PresentContentSize(ctx)
+			if err != nil {
+				return err
+			}
+			diskUsage += size
+			return nil
+		}); err != nil {
+			return 0, err
+		}
+	}
```

Comment:
This will sum the same blobs multiple times if they are shared by different images.
We should walk each content blob directly and then deduplicate based on visited content blobs.

Reply:
Fixed.